homeassistant/components/homekit/util.py

```diff
@@ -49,6 +52,15 @@ def validate_entity_config(values):
                         'Invalid mode: "{}", valid modes are: "{}".'
                         .format(key, MEDIA_PLAYER_MODES))
 
+        if domain == 'switch':
+            switch_type = config.get(CONF_TYPE)
+            if switch_type:
+                params[CONF_TYPE] = cv.string(switch_type)
+                if params[CONF_TYPE] not in SWITCH_TYPES:
+                    raise vol.Invalid(
+                        'Invalid type: "{}", valid types are: "{}".'
+                        .format(params[CONF_TYPE], SWITCH_TYPES))
```

Comment:
This section will probably change during the rebasing. Take a look at how I did the media_player validation above afterwards.